ingestion/src/metadata/ingestion/source/pipeline/databrickspipeline/metadata.py

```diff
@@ -0,0 +1,222 @@
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.models.pipeline_status import OMetaPipelineStatus
+from metadata.ingestion.source.database.databricks.client import DatabricksClient
+from metadata.ingestion.source.pipeline.pipeline_service import PipelineServiceSource
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+STATUS_MAP = {
+    "SUCCESS": StatusType.Successful,
+    "FAILED": StatusType.Failed,
+    "TIMEOUT": StatusType.Failed,
+    "CANCELED": StatusType.Failed,
+    "PENDING": StatusType.Pending,
+    "RUNNING": StatusType.Pending,
+    "TERMINATING": StatusType.Pending,
+    "SKIPPED": StatusType.Failed,
+    "INTERNAL_ERROR": StatusType.Failed,
+}
+
+TASK_MAP = {
+    "notebook_task": "Notebook_Task",
+    "spark_jar_task": "SparkJar_Task",
+    "spark_python_task": "SparkPython_Task",
+    "spark_submit_task": "SparkSubmit_Task",
+    "pipeline_task": "DeltaLiveTablePipeline_Task",
+    "python_wheel_task": "PythonWheel_Task",
+    "sql_task": "Sql_Task",
+    "dbt_task": "Dbt_Task",
+}
+
+
+class DatabrickspipelineSource(PipelineServiceSource):
+    """
+    Implements the necessary methods ot extract
+    Pipeline metadata from Databricks Jobs API
+    """
+
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        super().__init__(config, metadata_config)
+        self.task_id_mapping = {}
+        self.job_id_list = set()
+        self.connection = self.config.serviceConnection.__root__.config
+        self.client = DatabricksClient(self.connection)
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        """Create class instance"""
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: DatabricksPipelineConnection = (
+            config.serviceConnection.__root__.config
+        )
+        if not isinstance(connection, DatabricksPipelineConnection):
+            raise InvalidSourceException(
+                f"Expected DatabricksPipelineConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_pipelines_list(self) -> Iterable[dict]:
+        """
+        Get List of all pipelines
+        """
+        for workflow in self.client.list_jobs():
+            yield workflow
+
+    def get_pipeline_name(self, pipeline_details: dict) -> str:
+        """
+        Get Pipeline Name
+        """
+        return pipeline_details["settings"]["name"]
+
+    def yield_pipeline(self, pipeline_details: Any) -> Iterable[CreatePipelineRequest]:
+        """
+        Method to Get Pipeline Entity
+        """
+        self.job_id_list = set()
+        pipeline_ev = CreatePipelineRequest(
```

Comment:
maybe we want to try/catch this call. I see we also have the `get_tasks` function in there without handling exceptions. This could break the workflow execution. Usually it's ok to skip an asset of being ingested, but we don't want the whole workflow to die

Reply:
Updated with a try/catch. I mirrored the approach used in the AirflowSource pipeline service.

Reply:
thanks, then we might need to improve the process in airflow too hehe. In Airflow I feel a bit more comfortable as the data we pick up is from a db instead of an API, so we have a bit more safety once the connection is done. In any case, it should be handled